fe/fe-core/src/main/java/com/starrocks/transaction/PublishVersionDaemon.java

```diff
@@ -251,69 +271,107 @@ boolean isLakeTableTransaction(TransactionState transactionState) {
         return false;
     }
 
-    // todo: refine performance
-    void publishVersionForLakeTable(List<TransactionState> readyTransactionStates) throws UserException {
-        GlobalTransactionMgr globalTransactionMgr = GlobalStateMgr.getCurrentGlobalTransactionMgr();
-
+    void publishVersionForLakeTable(List<TransactionState> readyTransactionStates) {
+        int maxPublishingTransactions = Config.experimental_lake_publish_version_threads * 2;
```

Comment:
A random chosen number

Reply:
A transaction state may contain multiple tables and multiple partitions, it may be useless to process more states than `experimental_lake_publish_version_threads`.

Reply:
The scheduling period of `PublishVersionDaemon` is 10ms, submitting more tasks might increase the thread pool usage, i.e, try to keep the thread pool busy during the 10ms scheduling cycle.